src/shared/Core/ProcessManager.cs

```diff
@@ -50,9 +57,53 @@ public virtual ChildProcess CreateProcess(string path, string args, bool useShel
         return CreateProcess(psi);
     }
 
-
     public virtual ChildProcess CreateProcess(ProcessStartInfo psi)
     {
         return new ChildProcess(Trace2, psi);
     }
+
+    /// <summary>
+    /// Create a TRACE2 "session id" (sid) for this process.
+    /// </summary>
+    public static void CreateSid()
+    {
+        Sid = Environment.GetEnvironmentVariable(SidEnvar);
+
+        if (!string.IsNullOrEmpty(Sid))
+        {
+            Sid = $"{Sid}/{Guid.NewGuid():D}";
+        }
+        else
+        {
+            // We are the root process; create our own 'root' SID
+            Sid = Guid.NewGuid().ToString("D");
+        }
+
+        Environment.SetEnvironmentVariable(SidEnvar, Sid);
+        Depth = GetProcessDepth();
+    }
+
+    /// <summary>
+    /// Get "depth" of current process relative to top-level GCM process.
+    /// </summary>
+    /// <returns>Depth of current process.</returns>
+    internal static int GetProcessDepth()
+    {
+        char processSeparator = '/';
+        // Convert to string for .NET Framework compatibility.
+        if (!Sid.Contains(processSeparator.ToString()))
```

Comment:
nit:  Is this test necessary?  it seems like it just adds a second pass. 

That is, when Trace2 is enabled, GCM will almost always be called by Git with a valid SID and we'll already have created our own SID with a slash.  So this test will fail and we'll have to do the second AsSpan() loop anyway.

It feels like we can just omit this and just do the AsSpan() loop and save an extra scan on the string.

Reply:
Good point!